src/main/java/com/cloudbees/jenkins/support/filter/NameProvider.java

```diff
@@ -0,0 +1,134 @@
+    private final Supplier<Stream<String>> names;
+    private final Supplier<String> fakes;
+
+    private NameProvider(@Nonnull Supplier<Stream<String>> names, @Nonnull Supplier<String> fakes) {
+        this.names = names;
+        this.fakes = fakes;
+    }
+
+    /**
+     * @return a stream of names to anonymize
+     */
+    public @Nonnull Stream<String> names() {
+        return names.get();
+    }
+
+    /**
+     * @return a new fake name to use for anonymization
+     */
+    public @Nonnull String generateFake() {
+        return fakes.get();
+    }
+
+    /**
+     * @return all registered NameProviders
+     */
+    public static @Nonnull ExtensionList<NameProvider> all() {
+        return ExtensionList.lookup(NameProvider.class);
+    }
+
+    /**
+     * Provides the names of items.
+     */
+    public static final @Extension NameProvider ITEMS = new NameProvider(
+            () -> Jenkins.get()
+                    .getItems().stream()
+                    .flatMap(item -> Arrays.stream(item.getFullName().split("/")))
+                    .distinct(),
+            DataFaker.get().apply(faker -> "item_" + faker.app().name()));
+
+    /**
+     * Provides the names of view.
+     */
+    public static final @Extension NameProvider VIEWS = new NameProvider(
+            () -> Jenkins.get()
+                    .getViews().stream()
+                    .map(View::getViewName),
+            DataFaker.get().apply(faker -> "view_" + faker.space().moon()));
+
+    /**
+     * Provides the names of nodes.
+     */
+    public static final @Extension NameProvider NODES = new NameProvider(
+            () -> Jenkins.get()
+                    .getNodes().stream()
+                    .map(Node::getNodeName),
+            DataFaker.get().apply(faker -> "node_" + faker.internet().domainWord()));
+
+    /**
+     * Provides the names of computers.
+     */
+    public static final @Extension NameProvider COMPUTERS = new NameProvider(
+            () -> Arrays.stream(Jenkins.get().getComputers())
+                    .map(Computer::getDisplayName),
+            DataFaker.get().apply(faker -> "computer_" + faker.internet().domainWord()));
+
+    /**
+     * Provides the names of users.
+     */
+    public static final @Extension NameProvider USERS = new NameProvider(
+            () -> User.getAll().stream()
+                    .map(User::getFullName),
+            DataFaker.get().apply(faker -> "user_" + faker.name().username()));
+
+    /**
+     * Provides the names of labels. Note that this extension is given a lower priority than the others to avoid
+     * naming conflicts between labels and nodes/computers.
+     */
+    public static final @Extension(ordinal = -100) NameProvider LABELS = new NameProvider(
+            () -> Jenkins.get()
+                    .getLabels().stream()
+                    .map(Label::getDisplayName),
+            DataFaker.get().apply(faker -> "label_" + faker.space().star()));
```

Comment:
I don't have a strong opinion about the classes of fake names, but it might be better to use the same class here as you do for nodes and computers since they are used in similar contexts in Jenkins.

Reply:
Makes sense to me. I didn't think of that at the time.